testcases/OpTestHMIHandling.py

```diff
@@ -502,15 +517,12 @@ def runTest(self):
 class MalfunctionAlert(OpTestHMIHandling):
     def runTest(self):
         self._testHMIHandling(BMC_CONST.HMI_MALFUNCTION_ALERT)
```

Comment:
I see MalfunctionAlert is not working as expected..
Ideally after error injection system will re-IPL with reporting a platform error, we just need to wait for IPL to start and finish, Looks like with this changes it is doing a another force IPL.

Reply:
Can you provide recreate ?  The cases I saw the SBE Update caused a few bounces.

Reply:
This below changes on top of your PR, made it worked as expected. so please add these changes.
```
diff --git a/testcases/OpTestHMIHandling.py b/testcases/OpTestHMIHandling.py
index 1260556..33aa69b 100644
--- a/testcases/OpTestHMIHandling.py
+++ b/testcases/OpTestHMIHandling.py
@@ -49,7 +49,7 @@ from common.OpTestSystem import OpSystemState
 from common.OpTestSSH import ConsoleState as SSHConnectionState
 from common.OpTestIPMI import IPMIConsoleState
 from common.OpTestConstants import OpTestConstants as BMC_CONST
-from common.Exceptions import CommandFailed, UnknownStateTransition, PlatformError, HostbootShutdown
+from common.Exceptions import CommandFailed, UnknownStateTransition, PlatformError, HostbootShutdown, StoppingSystem
 
 class OpTestHMIHandling(unittest.TestCase):
     @classmethod
@@ -63,7 +63,7 @@ class OpTestHMIHandling(unittest.TestCase):
         cls.util = OpTestUtil()
 
     def setUp(self):
-        self.handle_ipl()
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
         self.cv_HOST.host_enable_all_cores()
 
     def clear_stop(self):
@@ -71,15 +71,20 @@ class OpTestHMIHandling(unittest.TestCase):
         self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
 
     def handle_ipl(self):
-        try:
-            self.cv_SYSTEM.goto_state(OpSystemState.OS)
-        except (UnknownStateTransition, PlatformError, HostbootShutdown) as e:
-            print "OpTestHMIHandling handle_ipl Exception={}".format(e)
-            self.cv_SYSTEM.stop = 0
+        for i in range(3):
+            try:
+                self.cv_SYSTEM.set_state(OpSystemState.IPLing)
+                self.cv_SYSTEM.goto_state(OpSystemState.OS)
+                break
+            except (UnknownStateTransition, PlatformError, HostbootShutdown, StoppingSystem) as e:
+                print "OpTestHMIHandling handle_ipl Exception={}".format(e)
+                self.clear_stop()
+        else:
+            self.clear_stop()
             self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
             try:
                 self.cv_SYSTEM.goto_state(OpSystemState.OS)
-            except (UnknownStateTransition, PlatformError, HostbootShutdown) as e:
+            except (UnknownStateTransition, PlatformError, HostbootShutdown, StoppingSystem) as e:
                 print "OpTestHMIHandling handle_ipl secondary Exception={}".format(e)
                 self.cv_SYSTEM.stop = 0
                 self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
@@ -360,12 +365,11 @@ class OpTestHMIHandling(unittest.TestCase):
 
         # Core checkstop will lead to system IPL, so we will wait for certain time for IPL
         # to finish
-        console = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        console = self.cv_SYSTEM.console
         res = console.run_command("uname -a") # perform any command to make sure console is logged in
 
         # now can send raw pexpect commands which assume log in
         console.sol.sendline(l_cmd)
-        self.clear_stop()
         self.handle_ipl()
 
     ##
@@ -394,7 +398,6 @@ class OpTestHMIHandling(unittest.TestCase):
 
         # now can send raw pexpect commands which assume log in
         console.sol.sendline(l_cmd)
-        self.clear_stop()
         self.handle_ipl()
 
     ##
```

Reply:
> for i in range(3):

Any reason 3 was chosen ?

Reply:
yes, after error injection, if SBE updates take place due to changes in configuration of cores, upto max 3 re-ipl's will take place. If 3rd attempt fails that will be actual IPL failure. So we need to allow system to ipl itself successfully within 3 attempts.

Reply:
> def setUp(self): - self.handle_ipl() + self.cv_SYSTEM.goto_state(OpSystemState.OS) self.cv_HOST.host_enable_all_cores()

Can you please give a recreate on why the goto_state OS is needed coming out of handle_ipl ?  It seems that if handle_ipl is working that state should be OS ?

I had a test case flow when the system was left in a bad state, so handle_ipl is needed to properly catch the unexpected state that maybe someone left the machine in.

Reply:
I will provide new patch that I believe will cover unexpected cases and such.  The logic in handle_ipl will allow the proper cycles for SBE Update and then fail if its a true failure.

If with the new patch level it is still a problem, we can debug a recreate to understand better.

Reply:
Sure, Please push your new changes. I will test it. Thanks.

Reply:
> Please push your new changes

Pushed latest changes, I think I captured all the changes.

Reply:
> Please push your new changes

Pushed latest fixes to add check for ISTEP.
Still working on failure flows when ISTEP does not happen.
Noticed on wspoon8 that Petitboot lost its boot order post SBE Updates, so I had to manually catch the console to set the proper sdb2 device setup.

Reply:
> failure flows when ISTEP

testcases/OpTestHMIHandling.py

Reply:
Thanks, Looks good., Tests are now working as designed. Once it finishes, will merge it.